File: js/src/resources/balance.ts
import type { Transport } from "../http.js";
import type { Balance } from "../models.js";

export interface GetBalanceParams {
  format?: "full" | "short";
  currency?: string;
}

export class BalanceResource {
  constructor(private readonly t: Transport) {}

  get(params: GetBalanceParams = {}): Promise<Balance> {
    return this.t.request<Balance>("GET", "/sms/balance", {
      query: { format: params.format ?? "full", currency: params.currency },
    });
  }
}

File: python/callisto_sdk/resources/balance.py
from __future__ import annotations

from typing import Optional

from .._http import Transport
from ..models import Balance


class BalanceResource:
    def __init__(self, transport: Transport):
        self._t = transport

    def get(self, format: str = "full", currency: Optional[str] = None) -> Balance:
        data = self._t.request(
            "GET", "/sms/balance", query={"format": format, "currency": currency}
        )
        return Balance.from_dict(data)
